File: discord-music-bot/src/main/java/net/masterzach32/swagbot/commands/CommandEvent.java
package net.masterzach32.swagbot.commands;

import com.mashape.unirest.http.exceptions.UnirestException;
import sx.blah.discord.handle.obj.IMessage;
import sx.blah.discord.util.DiscordException;
import sx.blah.discord.util.MissingPermissionsException;
import sx.blah.discord.util.RateLimitException;

public interface CommandEvent {

	public void execute(IMessage message, String[] params) throws RateLimitException, MissingPermissionsException, DiscordException, UnirestException;
	
}

File: discord-music-bot/src/main/java/net/masterzach32/swagbot/commands/CommandEvent.kt
package net.masterzach32.swagbot.commands

import com.mashape.unirest.http.exceptions.UnirestException
import sx.blah.discord.handle.obj.IMessage
import sx.blah.discord.util.DiscordException
import sx.blah.discord.util.MissingPermissionsException
import sx.blah.discord.util.RateLimitException

import java.io.IOException

interface CommandEvent {

    @Throws(RateLimitException::class, MissingPermissionsException::class, DiscordException::class, UnirestException::class, IOException::class)
    fun execute(message: IMessage, params: Array<String>)

}